portals/publisher/source/src/app/components/Apis/Details/Endpoints/GeneralConfiguration/CertificateUsage.jsx

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, {useEffect, useState} from 'react';
+import {FormattedMessage, injectIntl} from 'react-intl';
+import {makeStyles, withStyles} from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import {Typography} from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import UsageIcon from '@material-ui/icons/List';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import API from 'AppData/api';
+import MUIDataTable from "mui-datatables";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+const styles = {
+    appBar: {
+        position: 'relative',
+    },
+    flex: {
+        flex: 1,
+    },
+    popupHeader: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+    },
+    splitWrapper: {
+        padding: 0,
+    },
+    docName: {
+        alignItems: 'center',
+        display: 'flex',
+    },
+    button: {
+        height: 30,
+        marginLeft: 30,
+    },
+};
+
+const useStyles = makeStyles(() => ({
+    root: {
+        width: '100%',
+        flexDirection: 'row',
+        display: 'flex',
+    },
+    usageDialogHeader: {
+        fontWeight: '600',
+        fontSize: 'h6.fontSize',
+        marginRight: 10,
+    },
+    buttonIcon: {
+        marginRight: 10,
+    },
+}));
+
+/**
+ *
+ * @param {any} props Props for usage function.
+ * @returns {any} Returns the rendered UI for scope usage.
+ */
+function CertificateUsage(props) {
+    const { certAlias } = props;
+    const classes = useStyles();
+    const [open, setOpen] = useState(false);
+    const [usageData, setUsageData] = useState( []);
+    const [page, setPage] = useState(0);
+    const [rowsPerPage, setRowsPerPage] = useState(5);
+    const [count, setCount] = useState(-1);
```

Comment:
why do we set the initial count to -1?

Reply:
Because we don't know the exact count initially and that's how we mark it in the MUIDataTable component